proposals/lip-0051.md

```diff
@@ -2460,7 +2461,8 @@ storeValue = encode(
             "module": lockedBalance.module,
             "amount": lockedBalance.amount
         } for each lockedBalance in userEntry.lockedBalances]
-)
+    }
+)    
 ```
 
```

Comment:
Comment about line 2486: 

While creating entries in the supported tokens substore, in case `chainID = EMPTY_BYTES`, in theory the value of `supportedTokenIDs` could be an array of arbitrary long size containing tokenIDs. We should not write this in the state, instead we should use as value the empty array.

I see 2 possible ways to achieve this. 

1.  Either add an if condition in line 2488 to distinguish the case where `supportedTokensEntry.chainID = EMPTY_BYTES` or
2. Add an extra requirement in the verification above (e.g., line 2451) that the unique entry of the `genesisBlockAssetObject.supportedTokensSubstore` array with length 0 should have `entry.supportedTokenIDs = []`.

Reply:
I would rather write this in the verification section. I suggest the following, please check:
```
The `supportedTokensSubstore` array must either:
    - Have a unique entry with `chainID` of length `0` and no other entries. This is the case when all tokens are supported, i.e., `chainID == ALL_SUPPORTED_TOKENS_KEY`. In that case, the the `supportedTokenIds` is an empty array.
    - Have all entries with `chainID` property of length `CHAIN_ID_LENGTH` and `supportedTokenIDs` array in lexicographic order. 
```

Reply:
Sounds good. Syntactically, we should use either ... or ...

For the second bullet point don't forget to copy/paste that the `supportedTokensSubstore` array should be sorted by lexicographic order of `chainID`.